packages/vue-components/src/Navbar.vue

```diff
@@ -71,12 +75,114 @@
         this.collapsed = !this.collapsed
+      },
+      normalizeUrl(url) {
+        if (url.endsWith('.html')) {
+          return url.toLowerCase();
+        } else if (url.endsWith('/')) {
+          return `${url}index.html`.toLowerCase();
+        } else if (!url.endsWith('/')) {
+          return `${url}.html`.toLowerCase();
+        } else {
+          return url.toLowerCase();
+        }
+      },
+      // Splits a normalised URL into its parts, e.g http://site.org/foo/bar/index.html -> ['foo','bar','index.html']
+      splitUrl(url) {
+        const u = new URL(this.normalizeUrl(url));
+        return `${u.pathname}`.substr(1).split('/');
+      },
+      isSibling(url, href) {
+        const hParts = this.splitUrl(href);
+        const uParts = this.splitUrl(url);
+        if (hParts.length !== uParts.length) {
+          return false;
+        }
+        for (let i = 0; i < hParts.length - 1; i++) {
+          if (hParts[i] !== uParts[i]) {
+            return false;
+          }
+        }
+        return true;
+      },
+      isChild(url, href) {
+        const hParts = this.splitUrl(href);
+        const uParts = this.splitUrl(url);
+        if (uParts.length <= hParts.length) {
+          return false;
+        }
+        for (let i = 0; i < hParts.length; i++) {
+          if (hParts[i] !== uParts[i]) {
+            return false;
+          }
+        }
+        return true;
+      },
+      isExact(url, href) {
+        return this.normalizeUrl(url) === this.normalizeUrl(href);
+      },
+      highlightLink(url) {
+        const defHlMode = this.defaultHighlightOn;
+        const navLis = Array.from(this.$el.querySelector('.navbar-nav').children);
+        // attempt an exact match first
+        for (const li of navLis) {
+          const navLinks = Array.from(li.querySelectorAll('a.nav-link'));
+          const dropdownLinks = Array.from(li.querySelectorAll('a.dropdown-item'));
+          const allNavLinks = navLinks.concat(dropdownLinks).filter(a => a.href);
+          for (const a of allNavLinks) {
+            const hlMode = a.getAttribute('data-highlight') || defHlMode;
+            if (hlMode === 'none') {
+              continue;
+            }
+            // terminate early on an exact match
+            if (this.isExact(url, a.href)) {
+              li.classList.add('current');
+              return;
+            }
+          }
+        }
+        // fallback to user preference, otherwise
+        for (const li of navLis) {
+          const navLinks = Array.from(li.querySelectorAll('a.nav-link'));
+          const dropdownLinks = Array.from(li.querySelectorAll('a.dropdown-item'));
+          const allNavLinks = navLinks.concat(dropdownLinks).filter(a => a.href);
+          for (const a of allNavLinks) {
+            const hlMode = a.getAttribute('data-highlight') || defHlMode;
+            if (hlMode === 'none') {
+              continue;
+            }
+            if (hlMode === 'exact') {
```

Comment:
is this needed with the one above? (let's keep the one above if not since all exact checks should come first)